### PR TITLE
Implement Textbook demo feedback

### LINKED ISSUE #1473: Implement feedback on Textbook demo page

### PR DESCRIPTION
**Pull request by y4izus**
Fix #1473

### CODE REVIEW

**Comment by JRussellHuffman** [comment]
The dropdown implementation looks really great. My only feedback is that for the medium size, I think it will be best if the dropdown still has an internal scroll like the large size. It makes perfect sense on the small size to not have the scroll.

File: components/textbook-demo/TextbookDemoHeader.vue [hunk 1]
```diff
@@ -5,21 +5,22 @@
...
         </AppPageHeaderTitle>
-        <div class="textbook-demo-header__dropdown-container">
-          <AppMegaDropdownMenu />
-        </div>
-      </div>
-      <div class="textbook-demo-header__cta-container">
-        <AppCta v-bind="cta" />
+        <AppMegaDropdownMenu
+          class="textbook-demo-header__dropdown"
+          :placeholder="dropdownMenuPlaceholder"
```
Comment (techtolentino) [inline components/textbook-demo/TextbookDemoHeader.vue:10]: I like the placeholder 👍

File: constants/megaMenuLinks.ts [hunk 2]
```diff
@@ -1,351 +1,338 @@
...
 
-type Group = NavLink[]
-const GROUP_1_1: Group = [
-  {
-    label: 'Prequisites',
-    url: 'https://qiskit.org/textbook/ch-prerequisites/setting-the-environment.html'
-  },
-  {
-    label: 'Setting the Environment',
+type MegaDropdownMenuGroup = {
```
Comment (techtolentino) [inline constants/megaMenuLinks.ts:3]: this whole refactor is very clean and easier to understand, thank you

**Comment by JRussellHuffman** [comment]
One other thing, can the dropdown be closed when a user clicks off of it? In other words, while the dropdown is open, clicking anywhere on the page that isn't the dropdown, will close it.

**Comment by y4izus** [comment]
Close the dropdown by clicking outside the component needs more research, I've created [this issue](https://github.com/Qiskit/qiskit.org/issues/1489) to do it in the future and not to block this PR

**Review by eddybrando — APPROVED** [review]
Very nice work @y4izus! 😄

**Pull request merged**